language/vm/vm-runtime/vm-cache-map/src/arena.rs

```diff
@@ -37,9 +37,13 @@ impl<T> Arena<T> {
     pub fn alloc(&self, value: T) -> &mut T {
         let arena = self.inner.lock().expect("lock poisoned");
         let value = arena.alloc(value);
-        // Extend the lifetime of the value to that of the arena. typed_arena::Arena guarantees
-        // that the value will never be moved out from underneath, and this wrapper guarantees
-        // that the arena will not be dropped.
+        // UNSAFE CODE: This code requires auditing before modifications may land.
+        // JUSTIFICATION: This extends the lifetime of the returned reference
+        // to the lifetime of the arena. typed_arena's will not move values
+        // and will only deallocate when the arena goes out of scope. This
+        // unsafe cast is used by typed_arena itself (see the implementation
+        // of alloc_extend).
+        // AUDITOR: metajack
         unsafe { ::std::mem::transmute::<&mut T, &mut T>(value) }
```

Comment:
So this looks like one of those cases where the whole module is unsafe due to a single small use of unsafe?

Reply:
Pretty much. The safety depends on invariants throughout the module, as it requires that lifetimes be controlled. Thankfully the module is small and the invariant is pretty easy to keep.